contribute/design/cstor-operator/day2ops.md

```diff
@@ -0,0 +1,309 @@
+      blockDevices:
+      - blockDeviceName: pool-1-bd-3
+        id: pool-1-bd3
+        // System generated
+        capacity: 12345234
+
+      - blockDeviceName: pool-1-bd-4
+        id: pool-1-bd4
+        // System generated
+        capacity: 12345234
+
+  - name: gke-cstor-it-default-pool-2
+
+    groups:
+    - groupName: group1
+      blockDevices:
+      - blockDeviceName: pool-2-bd-1
+        // System generated
+        id: pool-2-bd1
+        // System generated
+        capacity: 12345234
+      - blockDeviceName: pool-2-bd-2
+        // System generated
+        id: pool-2-bd2
+        // System generated
+        capacity: 12345234
+
+    // New block device group added
+    - groupName: group2
+      blockDevices:
+      - blockDeviceName: pool-2-bd-3
+        // System generated
+        id: pool-2-bd3
+        // System generated
+        capacity: 12345234
+
+      - blockDeviceName: pool-2-bd-4
+        // System generated
+        id: pool-2-bd4
+        // System generated
+        capacity: 12345234
+
+```
+
+**NOTE:**
+In case of striped pool expansion no new group but block device in the present
+group as striped has only one groups but in all other raid groups a new group is
+added for expansion.
+
+### Operations Workflow
+( Includes Pool Expansion, Pool Deletion, Block Device Offline, Block Device Replacment and Pool Migration)
+
+Work done by CStor-operator after the user step is performed
+
+1. The node spec hash will change as a result of user adding/removing
+   blockDevices to CSPC or changing hostName for pool and then figure out what
+   operation it is!
+
+2. Host name changed in the CSPC can cause following:( Pool Migration)
+   Cstor-Operator will patch the corresponding CSP with the host name in the CSP
+   spec. Pool-mgmt should handle this change in CSP
+
+3. Block device removal can cause following:
+   i) Top level vdev not lost(Block Device Offline):
+      If a block device is removed ( e.g. one block device from a mirrored pool group),
+      the corresponding CSP is checked and block device that is present in CSP but not
+      present in CSPC is marked `false` by setting `inUseByPool` field of block
+      device on CSP to `false`. This change that occured in CSP should be
+      handled by pool-mgmt.
+
+   ii) Top level vdev lost (Pool Deletion):
+       If block devices from CSPC is removed in such a way that can
+       causes loss of top level vdev i.e. pool should be deleted)  (e.g. both
+       the block device in a mirrored group is removed) then the corresponding CSP for
+       which this has happened will be deleted. Obviously, when block devices
+       are removed `inUseByPool` field are marked `false`. And if Cstor-Operator
+       finds a number of block devices having `inUseByPool` set to `false` which causes
+       loss of top level vdev ( e.g. both the block device in a mirrored group are
+       marked false) then CStor-Operator deletes the corresponding CSP.
```

Comment:
Do we need to have any failsafe checks. e.g. if admin does it by mistake do we delete data of all the volumes on this pool? cc @vishnuitta